src/iio_wrapper.cpp

```diff
@@ -100,6 +100,48 @@ IIOWrapper::IIOWrapper()
         m_local_context = iio_create_local_context();
         if(!m_local_context)
             throw std::runtime_error("Exception: local context is null");
+
+        int count = iio_context_get_devices_count(m_local_context);
+        if(count>0)
+        {
+            struct iio_device *dev = iio_context_get_device(m_local_context, 0);
+            const char *name = iio_device_get_name(dev);
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_device_name"), "device name: %s", name);
+
+            IIOWrapper::s_device_name = std::string(name);
+
+            std::list<std::string> adis16505_x;
+            adis16505_x.push_back("adis16505");
+            adis16505_x.push_back("adis16505-1");
+            adis16505_x.push_back("adis16505-2");
+            adis16505_x.push_back("adis16505-3");
+
+            bool found = (std::find(adis16505_x.begin(), adis16505_x.end(),
+                                    IIOWrapper::s_device_name) != adis16505_x.end());
+            if(found)
+            {
+                IIOWrapper::s_device_name_enum = IIODeviceName::ADIS16505;
+            }
+
+            std::list<std::string> adis1657x;
+            adis1657x.push_back("adis16577");
+            adis1657x.push_back("adis16577-1");
+            adis1657x.push_back("adis16577-2");
+            adis1657x.push_back("adis16577-3");
+
+            bool found7x = (std::find(adis1657x.begin(), adis1657x.end(),
+                                      IIOWrapper::s_device_name) != adis1657x.end());
+            if(found7x)
+            {
+                IIOWrapper::s_device_name_enum = IIODeviceName::ADIS1657X;
+            }
+
+            if(!found && !found7x)
```

Comment:
Logically the two search loops can be merged together, if there is no other reason to keep them apart